core/controllers/acl_decorators_test.py

```diff
@@ -1343,6 +1343,85 @@ def test_non_author_cannot_resubmit_suggestion(self):
+    """Tests for get_decorator_for_accepting_suggestion decorator."""
+    owner_username = 'owner'
+    owner_email = 'owner@example.com'
+    author_username = 'author'
+    author_email = 'author@example.com'
+    username = 'user'
+    user_email = 'user@example.com'
+    TARGET_TYPE = 'exploration'
+    SUGGESTION_TYPE = 'edit_exploration_state_content'
+    exploration_id = 'exp_id'
+    target_version_id = 1
+    change_dict = {
+        'cmd': 'edit_state_property',
+        'property_name': 'content',
+        'state_name': 'Introduction',
+        'new_value': ''
+    }
+
+    class MockHandler(base.BaseHandler):
+        GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+        @acl_decorators.get_decorator_for_accepting_suggestion(
+            acl_decorators.can_edit_exploration)
+        def get(self, target_id, suggestion_id):
+            self.render_json({
+                'target_id': target_id,
+                'suggestion_id': suggestion_id
+            })
+
+    def setUp(self):
+        super(DecoratorForAcceptingSuggestionTests, self).setUp()
+        self.signup(self.author_email, self.author_username)
+        self.signup(self.user_email, self.username)
+        self.signup(self.owner_email, self.owner_username)
+        self.author_id = self.get_user_id_from_email(self.author_email)
+        self.owner_id = self.get_user_id_from_email(self.owner_email)
+        self.owner = user_services.UserActionsInfo(self.owner_id)
+        self.mock_testapp = webtest.TestApp(webapp2.WSGIApplication(
+            [webapp2.Route(
+                '/mock/<target_id>/<suggestion_id>', self.MockHandler)],
+            debug=feconf.DEBUG,
+        ))
+        self.save_new_default_exploration(self.exploration_id, self.owner_id)
+        rights_manager.publish_exploration(self.owner, self.exploration_id)
+        suggestion_services.create_suggestion(
+            self.SUGGESTION_TYPE, self.TARGET_TYPE,
+            self.exploration_id, self.target_version_id,
+            self.author_id,
+            self.change_dict, '', None)
+        suggestion = suggestion_services.query_suggestions(
+            [('author_id', self.author_id),
+             ('target_id', self.exploration_id)])[0]
+        self.suggestion_id = suggestion.suggestion_id
+
+    def test_guest_cannot_accept_suggestion(self):
+        with self.swap(self, 'testapp', self.mock_testapp):
+            self.get_json(
+                '/mock/%s/%s' % (self.exploration_id, self.suggestion_id),
+                expected_status_int=401)
+
+    def test_owner_can_accept_suggestion(self):
+        self.login(self.owner_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock/%s/%s' % (self.exploration_id, self.suggestion_id))
+        self.assertEqual(response['suggestion_id'], self.suggestion_id)
+        self.assertEqual(response['target_id'], self.exploration_id)
```

Comment:
Please also add a check that the content of the state has changed.

Reply:
The content will change only when the owner accepts the suggestion, but the above test only checks if the user _can_ _accept_ the suggestion and not _actually_ _accept_ it(hence the content won't be changed)

Reply:
Makes sense, sgtm.